quix-frontend/client/src/lib/sql-autocomplete/adapter/tests/test-utils/test-getCompletionItemsFromQueryContext.spec.ts

```diff
@@ -0,0 +1,110 @@
+
+import { DbInfoService } from "../../../../sql-autocomplete/db-info";
+import { ContextType, TableType } from "../../../sql-context-evaluator";
+import { SqlAutocompleter } from '../../sql-autocomplete-adapter';
+import {expect} from 'chai';
+import { MockDbInfoService } from "./mock-db-config";
+
+describe('testing autoComplete for nested objects:   ', () => {
+  const config = new MockDbInfoService();
+  const autocomp = new SqlAutocompleter(config);
+  const queryContext = {
+    prefix : '',
+    contextType: ContextType.Column,
+    tables : [
+      { type: TableType.External,
+        name: 'catalog0.schema0.TV&Sports',
+        alias: "",
+        selectAll : false,
+        tableRefs: [],
+        columns : [
+          {
+            name: 'TV',
+            dataType: 'row(movies row(action row(The_Dark_Knight varchar,The_Avengers varchar,Die_Hard varchar), comedy row(The_Hangover varchar, The_Hangover2 varchar)), tvSeries row(sopranos varchar, The_wire varchar))'
+          },
+          {
+            name: 'sports',
+            dataType: 'row(team row(basketball varchar, baseball varchar, soccer varchar), individual row(tennis varchar, racketBall varchar))'
+          },
+        ]
+      },
+      { type: TableType.External,
+        name: 'catalog1.schema1.food&countries',
+        alias: 'aliasName',
+        selectAll : false,
+        tableRefs: [],
+        columns : [
+          {
+            name: 'food',
+            dataType: 'row(fastFood row(mcDonalds row(hamBurger varchar,mcflurry varchar,french_fries varchar), asain row(sushi varchar, wok varchar)), local row(falafel varchar, hummus varchar))'
+          },
+          {
+            name: 'countries',
+            dataType: 'row(israel row(telAviv varchar, jerusalem varchar, ramatGan varchar), USA row(NYC varchar, california varchar))'
+          },
+        ]
+      }
+    ],
+  }
+  
+
+  it('should return movies and tv series for prefix catalog0.schema0.TV&Sports.TV.',async () => {
```

Comment:
The name of the tests (all of them) suppose to tell the developer **what** are you checking, not the input/output.